File: app/schemas/subscription.py
from datetime import datetime

from pydantic import BaseModel, Field


class SubscriptionBase(BaseModel):
    """Базовая схема подписки"""

    product_id: int = Field(..., gt=0, description="ID продукта")
    plan_id: int = Field(..., gt=0, description="ID плана")


class SubscriptionCreate(SubscriptionBase):
    """Схема создания подписки"""

    pass


class SubscriptionResponse(SubscriptionBase):
    """Схема ответа с данными подписки"""

    id: int
    user_id: int
    status: str
    start_date: datetime
    end_date: datetime
    auto_renew: bool
    requests_used: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class SubscriptionStatus(BaseModel):
    """Схема статуса подписки"""

    id: int
    status: str
    start_date: datetime
    end_date: datetime
    days_left: int
    requests_used: int
    auto_renew: bool

File: client/types/subscription.ts
export interface Subscription {
  id: number;
  user_id: number;
  product_id: number;
  plan_id: number;
  status: 'active' | 'expired' | 'cancelled' | 'suspended';
  start_date: string;
  end_date: string;
  auto_renew: boolean;
  requests_used: number;
  created_at: string;
  updated_at: string;
}

export interface SubscriptionStatus {
  id: number;
  status: string;
  start_date: string;
  end_date: string;
  days_left: number;
  requests_used: number;
  auto_renew: boolean;
}

export interface SubscriptionCreate {
  product_id: number;
  plan_id: number;
}

export interface SubscriptionWithDetails extends Subscription {
  product: {
    id: number;
    name: string;
    category: string;
  };
  plan: {
    id: number;
    name: string;
    price: number;
    duration_days: number;
  };
}
